fix evenloop and add neo4j

Paxwell-Paxwell opened the pull request:
Fix event loop handling:

Resolved the RuntimeError related to getting an event loop.
Changed the creation of event loops to ensure they run in the main thread using asyncio.get_event_loop()
Neo4j integration:

Added Neo4jKVStorage and Neo4jGraphStorage for improved graph-based data management.
Updated configuration to support Neo4j in LightRAG with necessary settings in the code.

Docker Compose:

Added a docker-compose.yml file to enable easy setup of the Neo4j database.

wiltshirek commented:
> Fix event loop handling:
> 
> Resolved the RuntimeError related to getting an event loop. Changed the creation of event loops to ensure they run in the main thread using asyncio.get_event_loop() Neo4j integration:
> 
> Added Neo4jKVStorage and Neo4jGraphStorage for improved graph-based data management. Updated configuration to support Neo4j in LightRAG with necessary settings in the code.
> 
> Docker Compose:
> 
> Added a docker-compose.yml file to enable easy setup of the Neo4j database.

Checking on this PR.  I've added a PR for Neo4J graph integration as well but I didn't add the KV storage yet.  Are you interested in merging that code?

Paxwell-Paxwell commented:
I’m interested, but it might take a bit of time. However, if you finish it quickly, feel free to go ahead.

Paxwell-Paxwell commented:
> > Fix event loop handling:
> 
> > 
> 
> > Resolved the RuntimeError related to getting an event loop. Changed the creation of event loops to ensure they run in the main thread using asyncio.get_event_loop() Neo4j integration:
> 
> > 
> 
> > Added Neo4jKVStorage and Neo4jGraphStorage for improved graph-based data management. Updated configuration to support Neo4j in LightRAG with necessary settings in the code.
> 
> > 
> 
> > Docker Compose:
> 
> > 
> 
> > Added a docker-compose.yml file to enable easy setup of the Neo4j database.
> 
> 
> 
> Checking on this PR.  I've added a PR for Neo4J graph integration as well but I didn't add the KV storage yet.  Are you interested in merging that code?  

I’m interested, but it might take a bit of time. However, if you finish it quickly, feel free to go ahead.

wiltshirek commented:
okay.  will do.  I'll keep you posted as things progress.  I'll leverage
what you have done so far if that makes sense to you.

On Sat, Nov 2, 2024 at 8:31 AM Pompom ***@***.***> wrote:

> Fix event loop handling:
>
> Resolved the RuntimeError related to getting an event loop. Changed the
> creation of event loops to ensure they run in the main thread using
> asyncio.get_event_loop() Neo4j integration:
>
> Added Neo4jKVStorage and Neo4jGraphStorage for improved graph-based data
> management. Updated configuration to support Neo4j in LightRAG with
> necessary settings in the code.
>
> Docker Compose:
>
> Added a docker-compose.yml file to enable easy setup of the Neo4j database.
>
> Checking on this PR. I've added a PR for Neo4J graph integration as well
> but I didn't add the KV storage yet. Are you interested in merging that
> code?
>
> I’m interested, but it might take a bit of time. However, if you finish it
> quickly, feel free to go ahead.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/HKUDS/LightRAG/pull/135#issuecomment-2452975984>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABG7F3ZHWZ6JQQOGCV2UPBLZ6TA3BAVCNFSM6AAAAABQTWFKTOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINJSHE3TKOJYGQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
*Ken Wiltshire** |* Executive Engineer
NYC Metro Area

Direct (347) 675-5213
